File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/05.While-Loop/03-additional/03.streamOfLetters.js
function streamOfLetters(input) {

    let index = 0;
    let current = input[index++];

    let secretWord = "";
    let secretMessage = "";
    let counterC = 0;
    let counterO = 0;
    let counterN = 0;
    let commandCounter = 0;

    while (current !== "End") {
        let symbolAsNum = current.charCodeAt();

        if ((symbolAsNum >= 65 && symbolAsNum <= 90) || (symbolAsNum >= 97 && symbolAsNum <= 122)) {
            let letter = current;

            if (letter === "c") {
                counterC++;
                if (counterC === 1) {
                    commandCounter++;
                } else {
                    secretWord += letter;
                }
            } else if (letter === "o") {
                counterO++;
                if (counterO === 1) {
                    commandCounter++;
                } else {
                    secretWord += letter;
                }
            } else if (letter === "n") {
                counterN++;
                if (counterN === 1) {
                    commandCounter++;
                } else {
                    secretWord += letter;
                }
            } else {
                secretWord += letter;
            }

            if (commandCounter === 3) {
                secretMessage += secretWord + " ";
                secretWord = "";
                counterC = 0;
                counterO = 0;
                counterN = 0;
                commandCounter = 0;
            }
        }

        current = input[index++];
    }

    console.log(secretMessage);
}
streamOfLetters(['o',
    'S',
    '%',
    'o',
    'l',
    '^',
    'v',
    'e',
    'c',
    'n',
    '&',
    'm',
    'e',
    'c',
    'o',
    'n',
    'End']);

File: Python_Basics/Homeworks-And-Labs/05.while_loop/03.additional/03.stream_of_letters.py
current = input()
secret_word = ""
secret_message = ""
counter_c = 0
counter_o = 0
counter_n = 0
command_counter = 0

while current != "End":
    symbol_as_num = ord(current)

    if (symbol_as_num >= 65 and symbol_as_num <= 90) or (symbol_as_num >= 97 and symbol_as_num <= 122):
        letter = current

        if letter == "c":
            counter_c += 1
            if counter_c == 1:
                command_counter += 1
            else:
                secret_word += letter
        elif letter == "o":
            counter_o += 1
            if counter_o == 1:
                command_counter += 1
            else:
                secret_word += letter
        elif letter == "n":
            counter_n += 1
            if counter_n == 1:
                command_counter += 1
            else:
                secret_word += letter
        else:
            secret_word += letter
        
        if command_counter == 3:
            secret_message += secret_word + " "
            secret_word = ""
            counter_c = 0
            counter_o = 0
            counter_n = 0
            command_counter = 0       
    current = input()

print(secret_message)